dipy/reconst/dki.py

```diff
@@ -1720,34 +1751,114 @@ def fit(self, data, mask=None):
             dki_params = params.reshape(out_shape)
             if self.return_S0_hat:
                 S0_params = S0_params.reshape(out_shape).squeeze()
+            if extra is not None:
+                for key in extra:
+                    self.extra[key] = extra[key].reshape(data.shape)
         else:
             dki_params = np.zeros(data.shape[:-1] + (27,))
             dki_params[mask, :] = params
             if self.return_S0_hat:
                 S0_params_in_mask = np.zeros(data.shape[:-1])
                 S0_params_in_mask[mask] = S0_params.squeeze()
                 S0_params = S0_params_in_mask
+            if extra is not None:
+                for key in extra:
+                    self.extra[key] = np.zeros(data.shape)
+                    self.extra[key][mask, :] = extra[key]
 
         return DiffusionKurtosisFit(self, dki_params, model_S0=S0_params)
 
     @multi_voxel_fit
-    def multi_fit(self, data_thres, mask=None):
+    def multi_fit(self, data_thres, mask=None, **kwargs):
+        """ Convenience function for fitting multiple voxels.
+        """
         extra_args = {} if not self.convexity_constraint else {
             'cvxpy_solver': self.cvxpy_solver,
             'sdp': self.sdp,
             }
+
         params, extra = self.fit_method(self.design_matrix, data_thres,
                                         self.inverse_design_matrix,
                                         return_S0_hat=self.return_S0_hat,
-                                        weights=self.weights,
                                         min_diffusivity=self.min_diffusivity,
-                                        **extra_args)
+                                        **extra_args,
+                                        **kwargs)
 
         S0_params = None
         if self.return_S0_hat:
             params, S0_params = params
 
-        return DiffusionKurtosisFit(self, params, model_S0=S0_params)
+        return DiffusionKurtosisFit(self, params, model_S0=S0_params), extra
+
+    def iterative_fit(self, data_thres, mask=None, num_iter=4,
+                      weights_method=weights_method_wls_gm):
```

Comment:
For all new function, we start to introduce keyword-only arguments, can you update ?

`def iterative_fit(self, data_thres, *, mask=None, num_iter=4, weights_method=weights_method_wls_gm):`

Reply:
I have been trying, what is the problem here? I haven't seen "*" like this, can you help explain please?

Reply:
I think, the best explanation is https://peps.python.org/pep-3102/

Reply:
Also, there no problem, we just enforced the use of keyword to simplify the maintenance and remove reordering argument issue.

Reply:
So adding the `*` after `data_thres` would accomplish this, but otherwise things are okay? i.e. `data_thres` still remains a positional argument?

Reply:
👍 yes

Reply:
I confirm, it is **ONLY** for new function.

we will take care of the old function with a decorator and a warning message